.github/workflows/main.yml

```diff
@@ -0,0 +1,109 @@
+name: CI
+
+on: [ push, pull_request ]
+
+jobs:
+  lockfile-lint:
+    name: Lockfile lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+      - name: lint lock file
+        run: npx lockfile-lint --path package-lock.json --allowed-hosts npm --validate-https
+
+  quality-checks:
+    name: Quality checks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+      - name: install dependencies
+        run: npm ci
+      - name: types check
+        run: npm run types-check
+      - name: lint check
+        run: npm run lint
+
+  test:
+    strategy:
+      matrix:
+        platform: [ ubuntu-latest ]
+        node: [ '10', '12', '14', '16' ]
+    needs: [ 'lockfile-lint', 'quality-checks' ]
```

Comment:
the reason we require `lockfile-lint` before running the test is is to check that the lock file isn't compromised before running `npm install`.

i would actually add `lockfile-lint` to be needed by `quality-checks`

Reply:
fixed